Makefile

```diff
@@ -84,7 +84,8 @@ check: gopath
 	@# Ensure no temp files are left behind
 	@$(eval BEFORESHA = $(shell ls -art /tmp | sha512sum))
 	go test -cover ${GO_PACKAGE_PREFIX}/...
-	@if [ "$(BEFORESHA)" != "$$(ls -art /tmp | sha512sum)" ] ; then \
+	@$(eval AFTERSHA = $(shell ls -art /tmp | sha512sum))
```

Comment:
Well, the problem here is that both the BEFORE and AFTER sha are calculated before any make targets are run since they are created with eval.
I ran into this in the original implementation and that's why I had to change it to be in the test.

Reply:
how come?